pkg/reconciler/vertex/controller.go

```diff
@@ -320,6 +320,7 @@ func (r *vertexReconciler) reconcile(ctx context.Context, vertex *dfv1.Vertex) (
 	vertex.Status.Selector = selector.String()
 
 	vertex.Status.MarkPhaseRunning()
+	vertex.Status.SetObservedGeneration(vertex.Generation)
```

Comment:
Set at the beginning?

Reply:
I have a follow up PR, will address it there.

Reply:
okay, Thanks